fix[Op-52158]: Drive id message should include `ID` - "Drive ID can't be blank"

akabiru opened the pull request:
#### https://community.openproject.org/wp/52158

Similar to https://github.com/opf/openproject/pull/14354/commits/ca0fc06d369ab3dc3c6d8a83290df1f612ba339b - active model errors strips out the "ID" from the message when the attribute name has an "_id" suffix. This overrides the default message by specifying the relation `drive` label

🐛  _Before_

![Screenshot 2024-01-17 at 3 30 00 PM](https://github.com/opf/openproject/assets/17295175/4ae0ab29-2985-4491-8349-75f7e6bb9120)

🔧  _After_

![Screenshot 2024-01-17 at 3 43 27 PM](https://github.com/opf/openproject/assets/17295175/4bbe00ee-5590-49a7-93ee-b279e36329c3)

akabiru committed:
fix[Op-51021]: drive id message should include `ID`

Kharonus commented on modules/storages/app/contracts/storages/storages/one_drive_contract.rb:
```diff
@@ -35,6 +35,8 @@ class OneDriveContract < ::ModelContract
     attribute :drive_id
-    validates :drive_id, presence: true, allow_nil: true
+    validates :drive_id,
+              presence: { message: I18n.t('activerecord.errors.models.storages/storage.attributes.drive_id.id_blank') },
```
❓ why do we need a specific localized message for `drive_id` and not for `tenant_id`? according to your screen, the message for tenant contains the id, but for drive it does not (before your fix). How is that?

akabiru replied:
active model errors strips out the "ID" from the message when the attribute name has an "_id" suffix.

akabiru replied:
> ❓ why do we need a specific localized message for `drive_id` and not for `tenant_id`? according to your screen, the message for tenant contains the id, but for drive it does not (before your fix). How is that?

Ah, indeed odd- let me compare! Could do without additional translations 👀

dominic-braeunlein reviewed (approved):
LGTM 👍

akabiru committed:
fix[Op-51021]: cascade attribute name change to oauth client id

Improve https://github.com/opf/openproject/commit/ca0fc06d369ab3dc3c6d8a83290df1f612ba339b

akabiru committed:
fix[Op-51021]: reinstate format

akabiru committed:
chore[Op#52158]: update error message

Kharonus reviewed (approved):
Brilliant! even removed localisations, we no longer need.
